packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1023,6 +1054,42 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
   private isValueEmpty = (value: ShorthandValue | ShorthandCollection) => {
     return _.isArray(value) ? value.length < 1 : !value
   }
+
+  private getHighlightedIndexOnArrowKeyOpen = (
+    changes: StateChangeOptions<ShorthandValue>,
+  ): number => {
+    // if open by ArrowUp, index should change by -1.
+    if (changes.type === Downshift.stateChangeTypes.keyDownArrowUp) {
+      const itemsLength = this.getItemsFilteredBySearchQuery().length
+      if (_.isNumber(this.state.highlightedIndex)) {
+        const newIndex = this.state.highlightedIndex - 1
+        return newIndex < 0 ? itemsLength - 1 : newIndex
+      }
+      return itemsLength - 1
+    }
+    // if open by ArrowDown, index should change by +1.
+    if (changes.type === Downshift.stateChangeTypes.keyDownArrowDown) {
+      const itemsLength = this.getItemsFilteredBySearchQuery().length
+      if (_.isNumber(this.state.highlightedIndex)) {
+        const newIndex = this.state.highlightedIndex + 1
+        return newIndex >= itemsLength ? 0 : newIndex
+      }
+      return 0
+    }
+    // if not opened by Arrow, index should not change at all.
+    return undefined
+  }
+
+  private getHighlightedIndexOnClose = (): number => {
+    if (!this.props.multiple && !this.props.search) {
+      // in single selection, if there is a selected item, highlight it.
+      if (this.state.value) {
+        return this.props.items.indexOf(this.state.value)
```

Comment:
it'd be nice to keep the index of the selected items in the state so that we don't have to perform a search every time; this would become something like:
```typescript
return this.state.selectedItemIndex // or this.state.valueIndex
```

Reply:
I don't know. that thing is used on selection. so once, twice maybe. then on each opening when dropdown is single selection, only when is already selected. once, twice maybe.

should we bother adding more complexity?

Reply:
then it's fine, no need to optimize, let's keep it simple, thx